### PR TITLE
🐛 Do not set last page as active if bookend is open

### LINKED ISSUE #18525: [amp-story] Video in last page will resume even when bookend is open
1. Use an amp-story with a video in the last page.
2. Navigate to open the bookend.
3. Go to another tab.
4. Return to the story.

Video will be playing in the background even when the bookend is open.

This is probably caused because `onPausedStateUpdate_` in `amp-story.js` is not checking if the activePage is behind an open bookend.

### PR DESCRIPTION
**Pull request by Enriqe**
Fixes #18525

Check if bookend is open before setting page as active.

### CODE REVIEW

**Comment by Enriqe** [comment]
That's true, I'll look into it.

**Comment by gmajoulet** [comment]
Enrique and I just talked offline:
It looks like we're not using the `PAUSED_STATE` when we open the bookend, but just pause the page manually. I think that's because we didn't have that `PAUSED_STATE` when we wrote the bookend's store logic.
Getting the `TOGGLE_BOOKEND` update both the `BOOKEND_STATE` and `PAUSED_STATE` seems like the right fix :)

Thanks Enrique!

**Comment by Enriqe** [comment]
Ready for another review :)

File: extensions/amp-story/1.0/test/test-amp-story.js [hunk 1]
```diff
@@ -164,6 +164,32 @@ describes.realWin('amp-story', {
         });
   });
 
+  it('should pause the story when bookend becomes active', () => {
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/test/test-amp-story.js]: Sorry to be picky about this, but I feel like these tests belongs in the `test-store-service` file.

To keep tests independent and a great coverage without too much complexity, I think what matters is that:
- We test that `TOGGLE_BOOKEND` also updates `PAUSED_STATE` ([example](https://github.com/ampproject/amphtml/blob/master/extensions/amp-story/1.0/test/test-amp-story-store-service.js#L167))
- Then, `PAUSED_STATE` and each component's behavior is tested separately. This logic you're testing is actually already well covered [here](https://github.com/ampproject/amphtml/blob/master/extensions/amp-story/1.0/test/test-amp-story.js#L533-L615)

This way, store's logic stays in the store's tests. And in `test-amp-story`, we simply test how the story component reacts to individual store state updates.

Comment (Enriqe) [reply extensions/amp-story/1.0/test/test-amp-story.js]: That's true... we should probably talk about our integration tests in general. e.g. When to write integration tests? Where should they live? Etc...

File: extensions/amp-story/1.0/test/test-amp-story-store-service.js [hunk 2]
```diff
@@ -172,6 +172,28 @@ describes.fakeWin('amp-story-store-service actions', {}, env => {
...
+
+    // First open the bookend.
+    storeService.dispatch(Action.TOGGLE_BOOKEND, true);
+
+    // Close the bookend.
+    storeService.dispatch(Action.TOGGLE_BOOKEND, false);
+
+    expect(pausedListenerSpy).to.have.been.calledTwice;
+    expect(pausedListenerSpy).to.have.been.calledWith(false);
+  });
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/test/test-amp-story-store-service.js]: Nit: you could register the `pausedListenerSpy` after you open the bookend, before you close it :)

**Pull request merged**